src/Stryker.CLI/Stryker.CLI/FileConfigReader.cs

```diff
@@ -0,0 +1,145 @@
+
+            inputs.WithBaselineInput.SuppliedInput =
+                config.Baseline is not null &&
+                (config.Baseline.Enabled.HasValue && config.Baseline.Enabled.Value);
+
+            inputs.BaselineProviderInput.SuppliedInput = config.Baseline?.Provider;
+            inputs.DiffIgnoreChangesInput.SuppliedInput = config.Since?.IgnoreChangesIn;
+            inputs.FallbackVersionInput.SuppliedInput = config.Baseline?.FallbackVersion;
+            inputs.AzureFileStorageUrlInput.SuppliedInput = config.Baseline?.AzureFileShareUrl;
+            inputs.CoverageAnalysisInput.SuppliedInput = config.CoverageAnalysis;
+            inputs.DisableBailInput.SuppliedInput = config.DisableBail;
+            inputs.DisableMixMutantsInput.SuppliedInput = config.DisableMixMutants;
+            inputs.AdditionalTimeoutInput.SuppliedInput = config.AdditionalTimeout;
+            inputs.MutateInput.SuppliedInput = config.Mutate;
+            inputs.MutationLevelInput.SuppliedInput = config.MutationLevel;
+            inputs.ProjectNameInput.SuppliedInput = config.ProjectInfo?.Name;
+            inputs.ModuleNameInput.SuppliedInput = config.ProjectInfo?.Module;
+            inputs.ProjectVersionInput.SuppliedInput = config.ProjectInfo?.Version;
+            inputs.ReportersInput.SuppliedInput = config.Reporters;
+
+            inputs.SinceTargetInput.SuppliedInput = config.Since?.Target;
+            inputs.SolutionInput.SuppliedInput = config.Solution;
+            inputs.TargetFrameworkInput.SuppliedInput = config.TargetFramework;
+
+            inputs.ProjectUnderTestNameInput.SuppliedInput = config.Project;
+            inputs.ThresholdBreakInput.SuppliedInput = config.Thresholds?.Break;
+            inputs.ThresholdHighInput.SuppliedInput = config.Thresholds?.High;
+            inputs.ThresholdLowInput.SuppliedInput = config.Thresholds?.Low;
+            inputs.VerbosityInput.SuppliedInput = config.Verbosity;
+            inputs.LanguageVersionInput.SuppliedInput = config.LanguageVersion;
+            inputs.TestProjectsInput.SuppliedInput = config.TestProjects;
+            inputs.TestCaseFilterInput.SuppliedInput = config.TestCaseFilter;
+            inputs.DashboardUrlInput.SuppliedInput = config.DashboardUrl;
+            inputs.IgnoreMutationsInput.SuppliedInput = config.IgnoreMutations;
+            inputs.IgnoredMethodsInput.SuppliedInput = config.IgnoreMethods;
+
+            inputs.ReportFileNameInput.SuppliedInput = config.ReportFileName;
+        }
+
+        private static FileBasedInput LoadConfig(string configFilePath)
+        {
+            using var streamReader = new StreamReader(configFilePath);
+            var fileContents = streamReader.ReadToEnd();
+
+            FileBasedInput input;
+            try
+            {
+                FileBasedInputOuter root;
+                if (configFilePath.EndsWith(".yaml") || configFilePath.EndsWith(".yml"))
```

Comment:
Probably this could be a case for a OOP exercise - like create some interface and some polymorphism :)

But either way thanks for this functionality.